Source/AST/Styling/Stylers/DownStyler.swift

```diff
@@ -0,0 +1,292 @@
+//
+//  DownStyler.swift
+//  Down
+//
+//  Created by John Nguyen on 22.06.19.
+//  Copyright © 2019 Glazed Donut, LLC. All rights reserved.
+//
+
+#if canImport(UIKit)
+
+import UIKit
+
+#elseif canImport(AppKit)
+
+import AppKit
+
+#endif
+
+/// A default `Styler` implementation that supports a variety of configurable
+/// properties for font, text color and paragraph styling, as well as formatting
+/// of nested lists and quotes.
+open class DownStyler: Styler {
+
+    // MARK: - Properties
+
+    public let fonts: FontCollection
+    public let colors: ColorCollection
+    public let paragraphStyles: ParagraphStyleCollection
+
+    public let quoteStripeOptions: QuoteStripeOptions
+    public let thematicBreakOptions: ThematicBreakOptions
+    public let codeBlockOptions: CodeBlockOptions
+
+    private let itemParagraphStyler: ListItemParagraphStyler
+
+    private var listPrefixAttributes: [NSAttributedString.Key : Any] {[
+        .font: fonts.listItemPrefix,
+        .foregroundColor: colors.listItemPrefix]
+    }
+
+    // MARK: - Init
+
+    public init(configuration: DownStylerConfiguration = DownStylerConfiguration()) {
+        fonts = configuration.fonts
+        colors = configuration.colors
+        paragraphStyles = configuration.paragraphStyles
+        quoteStripeOptions = configuration.quoteStripeOptions
+        thematicBreakOptions = configuration.thematicBreakOptions
+        codeBlockOptions = configuration.codeBlockOptions
+        itemParagraphStyler = ListItemParagraphStyler(options: configuration.listItemOptions, prefixFont: fonts.listItemPrefix)
+    }
+
+    // MARK: - Styling
+
+    open func style(document str: NSMutableAttributedString) {
+
```

Comment:
The methods with empty definitions are still being invoked by the `AttributedStringVisitor`. In this case, I did not to add any styling to the whole document. Styling for custom nodes (`CustomBlock`, `CustomInline`) however aren't being invoked because (I think) cmark doesn't actually support them (they're probably just stubs).

Anyway, the typical use case is not to invoke these methods directly, they should only be called by the styler. But I'm not sure we can enforce that.